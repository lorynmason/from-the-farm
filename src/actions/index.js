export const addUser = (user) => {
  return {
    type: 'ADD_USER',
    user
  };
};

export const removeUser = () => {
  return {
    type: 'REMOVE_USER'
  };
};

export const addVendors = (vendors) => {
  return {
    type: 'ADD_VENDORS',
    vendors
  };
};

export const addProducts = (products) => {
  return {
    type: 'ADD_PRODUCTS',
    products
  };
};

export const isLoading = (bool) => {
  return {
    type: 'IS_LOADING',
    bool
  };
};

export const hasErrored = (message) => {
  return {
    type: 'HAS_ERRORED',
    message
  };
};

export const addMessage = (message) => {
  return {
    type: 'ADD_MESSAGE',
    message
  };
};

export const addItemList = (items) => {
  return {
    type: 'ADD_ITEM_LIST',
    items
  };
};

export const vendorSearchResults = (results) => {
  return {
    type: 'VENDOR_SEARCH',
    results
  };
};

export const productSearchResults = (results) => {
  return {
    type: 'PRODUCT_SEARCH',
    results
  };
};

export const clearSearchResults = () => {
  return {
    type: 'CLEAR_SEARCH'
  };
};
