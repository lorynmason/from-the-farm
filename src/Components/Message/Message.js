import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addUser, removeUser, addProducts } from '../../actions';


export const Message = () => {
  
}

export const mapStateToProps = (state) => ({
  vendors: state.vendors,
  products: state.products,
  user: state.user
});

export const mapDispatchToProps = (dispatch) => {
  return {
    addUserToStore: (user) => dispatch(addUser(user)),
    fetchVendors: (url) => dispatch(fetchVendors(url)),
    addProductsToStore: (products) => dispatch(addProducts(products))
  } 
}

export default connect(mapStateToProps, mapDispatchToProps)(Message);