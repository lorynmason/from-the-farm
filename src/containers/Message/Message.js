import React from 'react';
import { connect } from 'react-redux';
import { addMessage } from '../../actions';
import PropTypes from 'prop-types';

export const Message = ({ message, addMessage}) => {
  setTimeout(() => {
    addMessage('')
  }, 5000);

  if(message) {
    return (
      <div className="message-container">
        <p>
          {message}
        </p>
      </div>
    );
  }
  
  return (
    <div className="message"></div>
  );
}

export const mapStateToProps = (state) => ({
  message: state.message
});

export const mapDispatchToProps = (dispatch) => ({
  addMessage: (message) => dispatch(addMessage(message))
});

export default connect(mapStateToProps, mapDispatchToProps)(Message);

Message.propTypes = {
  message: PropTypes.string,
  addMessage: PropTypes.func
}
