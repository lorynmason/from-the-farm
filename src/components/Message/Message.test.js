import React from 'react';
import { Message, mapDispatchToProps, mapStateToProps } from './Message';
import { shallow } from 'enzyme';
import { addMessage } from '../../actions';


describe('Message', () => {
  let mockFunc = jest.fn()
  let items = [{name: "Berries",
  id: 1}, {name: "Eggs",
  id: 11}]
  let user = {
    name: "Lemon Sisters",
    email: "regular@email.com",
    address: "12000 E. 47th Ave",
    phone: "(231) 341-4141",
    bio: "We Sell Lemons",
    lat: 39.7820535,
    long: -104.8503416,
    city: "denver",
    state: "CO",
    zip: 80239,
    img_url: null,
    token: "eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoyLCJleHAiOjE1NTA3MjM0OTZ9.az6Fs46yi1DL6u9J9cUWf3dt2aSbtsTHCNcTDGAQ-8k",
    products: [{name: "Broccoli",
    id: 9,
    user_id: 2,
    item_id: 2,
    price: 250,
    unit: "lb",
    description: "a lb of broccoli",
    vendorName: "Lemon Sisters"}]
  }
  let wrapper = shallow(<Message />)
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should match the snapshot if there is a message', () => {
    let message = 'There is a Messsage'
    wrapper = shallow(<Message message={message}/>)
    expect(wrapper).toMatchSnapshot()
  })
  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn()
    const mappedProps = mapDispatchToProps(mockDispatch)
  
    it('calls dispatch with an addMessage thunk when addMessage is called', () => {
      const expected = addMessage('You have a message');
      mappedProps.addMessage('You have a message')
      expect(mockDispatch).toHaveBeenCalledWith(expected)
    });
  })
})