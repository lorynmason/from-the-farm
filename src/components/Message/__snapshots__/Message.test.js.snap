// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Message) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Message should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "blah",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "blah",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Message should match the snapshot if there is a message 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message
    message="There is a Messsage"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        There is a Messsage
      </p>,
      "className": "message-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "There is a Messsage",
      },
      "ref": null,
      "rendered": "There is a Messsage",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          There is a Messsage
        </p>,
        "className": "message-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "There is a Messsage",
        },
        "ref": null,
        "rendered": "There is a Messsage",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
